docusaurus/docs/introduction/plugin-types-usage

```diff
@@ -0,0 +1,133 @@
+---
+id: plugin-types-usage
+title: Plugin types and usage
+description: Learn about the different types of Grafana plugins, their usage and configuration.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - usage
+  - provisioning
+  - provision
+  - configuration
+  - configure
+sidebar_position: 1
+---
+
+# Grafana plugin types and usage
+
+:::note
+
+This documentation discusses plugin configuration and usage for plugin developers. General information for plugin users is available at [Plugin management](https://grafana.com/docs/grafana/latest/administration/plugin-management/).
+
+:::
+
+Grafana plugin development allows for many options depending on the type of user experience you want to create. Make a plugin in one of several different types depending on whether you want to offer users a new way of visualizing data, a connection to a new database, or an integrated out-of-the-box experience.  
+
+The following table shows the plugin types currently supported by Grafana on the frontend and backend. You can also see whether or not each type is supported by the [Grafana API](https://grafana.com/api/plugins).
+
+|                |                 |                |                            |
+| -------------- | :-------------: | :------------: | :------------------------: |
+| **Type**       | **Frontend** |   **Backend**  | **SDK-supported** |
+| datasource     |       Yes       | Yes (optional) |             Yes            |
+| app            |       Yes       | Yes (optional) |             Yes            |
+| panel          |       Yes       |       No       |             Yes            |
+| renderer       |        No       |       Yes      |             No             |
+| secretsmanager |        No       |       Yes      |             No             |
+
+## Data source plugins
+
+In plugin development, you can create new data source plugins to use Grafana with new databases, similar to existing plugins such as [MongoDB](https://grafana.com/grafana/plugins/grafana-mongodb-datasource/) or [Google BigQuery](https://grafana.com/grafana/plugins/grafana-bigquery-datasource/).
+
+### Usage of data-source plugins
```

Comment:
I didn't think we used hyphens for this anywhere else?

Reply:
good question. see https://grafana.com/docs/writers-toolkit/write/style-guide/word-list/#d

Reply:
Reading that guide - aren't we referring to it within its noun form here?